### PR TITLE
wifi_popup

### PR DESCRIPTION
**Pull request by jankrzempek**
https://trello.com/c/uEXckYam/72-fixed-wifi-session-if-wifi-ssid-and-password-is-empty-user-cant-go-to-the-confirmationview-there-should-be-alert-informing-why

### CODE REVIEW

File: AirCasting/CreateSessionViews/CreateSessionDetailsView.swift [hunk 1]
```diff
@@ -47,54 +48,68 @@ struct CreateSessionDetailsView: View {
...
             }
             getAndSaveStartingLocation()
             isConfirmCreatingSessionActive = true
-            #warning("TODO: handle wifi empty password")
-            if wifiSSID != "" && wifiPassword != "" {
+            if wifiSSID != "", wifiPassword != "" {
                 sessionContext.wifiSSID = wifiSSID
                 sessionContext.wifiPassword = wifiPassword
+                showingAlert = false
+            } else if isWiFi, wifiSSID == "", wifiPassword == "" {
```
[the remark this answers is not in the document]
Comment (mateuszmackowiak) [reply AirCasting/CreateSessionViews/CreateSessionDetailsView.swift:75]: Seams like the check for empty is made few times. Imo it also is ok to have an empty wifiPassword. Some users don't have a wifipassword

File: AirCasting/CreateSessionViews/CreateSessionDetailsView.swift [hunk 2]
```diff
@@ -47,54 +48,68 @@ struct CreateSessionDetailsView: View {
...
-                })
+                }
         )
     }
 }
 
 private extension CreateSessionDetailsView {
     var continueButton: some View {
         Button(action: {
+            showingAlert = false
```
Comment (smialko) [inline AirCasting/CreateSessionViews/CreateSessionDetailsView.swift:63]: I think you don't have to do `showingAlert = false` here. You create it with an initial value so at this point it's still false. if we change it to true and then dismiss the alert it'll change the value back to false.

File: AirCasting/CreateSessionViews/CreateSessionDetailsView.swift [hunk 3]
```diff
@@ -47,54 +48,68 @@ struct CreateSessionDetailsView: View {
...
                 sessionContext.isIndoor = isIndoor
             }
             getAndSaveStartingLocation()
             isConfirmCreatingSessionActive = true
-            #warning("TODO: handle wifi empty password")
-            if wifiSSID != "" && wifiPassword != "" {
+            if wifiSSID != "", wifiPassword != "" {
                 sessionContext.wifiSSID = wifiSSID
                 sessionContext.wifiPassword = wifiPassword
+                showingAlert = false
```
Comment (smialko) [inline AirCasting/CreateSessionViews/CreateSessionDetailsView.swift:74]: Same as above :)

File: AirCasting/CreateSessionViews/WifiPopupView.swift [hunk 4]
```diff
@@ -29,9 +29,13 @@ struct WifiPopupView: View {
...
             VStack(spacing: 10) {
-                Button("Connect") {
-                    presentationMode.wrappedValue.dismiss()
-                } .buttonStyle(BlueButtonStyle())
+                if wifiSSID != "" && wifiPassword != "" {
+                    confirmationButton
+                } else {
+                    confirmationButton
+                    .accentColor(.gray)
+                    .disabled(true)
```
Comment (smialko) [inline AirCasting/CreateSessionViews/WifiPopupView.swift:37]: Considering what @mateuszmackowiak wrote, there's a possibility not to have a wifi password. So I guess we shouldn't disable connect button when the password is empty. :<

File: AirCasting/CreateSessionViews/WifiPopupView.swift [hunk 5]
```diff
@@ -66,6 +70,12 @@ struct WifiPopupView: View {
             isSSIDTextfieldDisplayed = true
         }
     }
+    
+    var confirmationButton: some View {
```
Comment (smialko) [inline AirCasting/CreateSessionViews/WifiPopupView.swift:74]: What do you think about renaming it to `connectButton`? 
I'm not good with the names, it's just an idea, please ignore it if you prefer it this way :D

File: AirCasting/CreateSessionViews/WifiPopupView.swift (hunk 4 above)
Comment (smialko) [reply AirCasting/CreateSessionViews/WifiPopupView.swift:37]: But you can leave some of the logic, and check `if wifiSSID.isEmpty` 🤔

File: AirCasting/CreateSessionViews/CreateSessionDetailsView.swift [hunk 6]
```diff
@@ -120,15 +135,15 @@ private extension CreateSessionDetailsView {
                 Text("Wi-Fi").tag(true)
             }
             .pickerStyle(SegmentedPickerStyle())
-            .onChange(of: isWiFi) { (_) in
+            .onChange(of: isWiFi) { _ in
```
Comment (smialko) [inline AirCasting/CreateSessionViews/CreateSessionDetailsView.swift:138]: nice catch 🙃

File: AirCasting/CreateSessionViews/CreateSessionDetailsView.swift [hunk 7]
```diff
@@ -47,54 +48,68 @@ struct CreateSessionDetailsView: View {
...
+            .alert(isPresented: $showingAlert, content: {
+                Alert(title: Text("Wi-Fi credentials are empty "),
+                      message: Text("Do you want to pop up Wi-Fi screen?"),
+                      primaryButton: .default(Text("Show Wi-fi screen")) {
+                          isWifiPopupPresented = true
+                      },
+                      secondaryButton: .default(Text("Cancel")))
+            })
+            .background(Group {
+                if (!isWiFi && wifiSSID == "" && wifiPassword == "") || (!isWiFi && wifiSSID != "" && wifiPassword != "") || (isWiFi && wifiSSID != "" && wifiPassword != "") {
```
Comment (smialko) [inline AirCasting/CreateSessionViews/CreateSessionDetailsView.swift:93]: I think we don't need it here. 
NavigationLink is inactive until `isConfirmCreatingSessionActive = true` and we check the wifi credentials in a button action. 🙂

**Review by smialko — CHANGES_REQUESTED** [review]
😎

**Comment by pawelgil** [comment]
Let's maybe squash commits (will require push -f) to leave one meaningful batch of changes for the sake of a clean git history?

**Comment by jankrzempek** [comment]
> Let's maybe squash commits (will require push -f) to leave one meaningful batch of changes for the sake of a clean git history?



> Let's maybe squash commits (will require push -f) to leave one meaningful batch of changes for the sake of a clean git history?

I did it, but for the first time and not sure if it's like it should be

**Review by smialko — APPROVED** [review]
👌🤠

**Pull request merged**